src/Common.cpp

```diff
@@ -208,6 +207,12 @@ std::string NAS2D::join(std::vector<std::string> strs, char delim, bool skip_emp
 	return result;
 }
 
+std::string NAS2D::trimWhitespace(std::string string)
+{
+	auto first_non_space = string.find_first_not_of(" \r\n\t\v\f");
+	auto last_non_space = string.find_last_not_of(" \r\n\t\v\f");
+	return string.substr(first_non_space, last_non_space - first_non_space + 1);
```

Comment:
I think this may have trouble with strings containing only whitespace. In that case you'd get:
`string.substr(-1, -1 - -1 + 1) = string.substr(-1, 1);`
With a first parameter of `-1`, [`substr`](https://en.cppreference.com/w/cpp/string/basic_string/substr) should thrown an `std::out_of_range` exception.

Instead, this method should return the empty string `""`.

Reply:
You are correct. This should return an empty string if the input contains only whitespace. Will add a check for `npos`.